src/OrleansRuntime/GrainDirectory/GrainDirectoryPartition.cs

```diff
@@ -364,21 +376,58 @@ internal void RemoveGrain(GrainId grain)
         internal AddressesAndTag LookUpActivations(GrainId grain)
```

Comment:
I agree - let's optimize those things iteratively, as separate PRs. I'm running a load test for this, for just in case. Will merge after that.